### ISSUE TITLE
compatibility with spaCy 3.0 ? 

### ISSUE THREAD

**Issue by d5555** [huggingface/neuralcoref#295]
Is there going  neuralcoref version compatible with spaCy 3.0 ?

**Comment by svlandeg** [comment]
Yes! This is definitely on our road map :-) I'll actually pin this issue because I'm sure others are interested as well.

**Comment by asterbini** [comment]
Plus 1

**Comment by asterbini** [comment]
After some rough paches I am able to start neuralcoref with spacy 3 but I obviously get the error 
''Trying to read a Model that was created with an incompatible version of Thinc''
Is there a way to convert the models to the new Thinc version used in Spacy 3?

**Comment by svlandeg** [comment]
@asterbini : Due to the many internal changes between Thinc 7 and 8, and spaCy 2 and 3, I'm not sure this is going to be feasible. See also [this discussion](https://github.com/explosion/spaCy/discussions/7054) at the spaCy forum.

We have now started working on a `coref` component native in spaCy v3 though. This won't be finished tomorrow, but we're making progress. We're going to reuse what we can from this library (with attribution, ofcourse) and make sure it trains and runs with native Thinc models :-)

**Comment by asterbini** [comment]
I was starting to look for a way to export the neuralcoref model in spacy2 as text, and try to load it in spacy3 :)
From your words this seems to be useless, thanks for letting me know.

**Comment by svlandeg** [comment]
I'm not a 100% sure though. If you have a working PyTorch model, that might operate with Thinc's new shim layers, but you'll have to write a custom component to wrap it in spaCy...

**Comment by asterbini** [comment]
But why the neuralcoref models should depend on the spacy3 models implementation? From a cursory glance to the NeuralCoref annotator class it seems to me that it reads only properties of the doc tokens that have not changed  from v2 to v3. Then unless Torch 8 is changed radically wrt 7, (and by mapping Relu to ReLu and Linear to Affine) perhaps there is a way to convert the binary data model to Torch 8.
(and yet, as I am a newbie of torch and neuralcoref an I am sure I could have missed something)

**Comment by KTRosenberg** [comment]
> @asterbini : Due to the many internal changes between Thinc 7 and 8, and spaCy 2 and 3, I'm not sure this is going to be feasible. See also [this discussion](https://github.com/explosion/spaCy/discussions/7054) at the spaCy forum.
> 
> We have now started working on a `coref` component native in spaCy v3 though. This won't be finished tomorrow, but we're making progress. We're going to reuse what we can from this library (with attribution, ofcourse) and make sure it trains and runs with native Thinc models :-)

Out of curiosity since I'm trying to integrate Spacy and Neuralcoref into a time-sensitive project, is the move to Spacy 3 going to take on the order of a couple of weeks, a couple of months, or a year? I'm not familiar with the amount of work necessary to do something like this.

**Comment by svlandeg** [comment]
I'd estimate a month, a couple of months at most.

**Comment by KTRosenberg** [comment]
Thanks! In that case, I might downgrade temporarily.

**Comment by d5555** [comment]
> I'd estimate a month, a couple of months at most.

That's great news! I can't wait :)

**Comment by ZhuoruLin** [comment]
Wow. Can't wait to use neuralcoref with SpaCy V3!

**Comment by jazzdup** [comment]
looking forward to this! thankyou!

**Comment by telmomenezes** [comment]
I'm staying with spaCy 2 until this comes out. Can't wait! Thanks for the great work!

**Comment by lauwauw** [comment]
Also looking forward to this! Hoping to use this in my project with the en_core_web_trf pipeline, which is dependent on spaCy 3. Thanks indeed for the great work so far!

**Comment by chinmaydas96** [comment]
This is an awesome project try on, waiting for the spacy 3.0 compartibility.

**Comment by grigvardanyan** [comment]
Hi @svlandeg, do you have approximate estimation when neuralcoref for spacy3 will be released ?
Thanks in advance

**Comment by Ar9av** [comment]
Following

**Comment by Yutong-FEI** [comment]
folloming

**Comment by seanswyi** [comment]
Also following.

**Comment by svlandeg** [comment]
Hi all, as a quick update that I also posted in a related thread.

While we (Explosion) originally intended to update `neuralcoref` in collaboration with HuggingFace (and @thomwolf had been very helpful with that), we've recently revamped spaCy to 3.0 and it turned out that quite a significant rewrite was necessary to support a native Thinc model in `neuralcoref`. Meanwhile also, research on coreference resolution had progressed further.

So long story short, we started working on a built-in coreference module that will be part of the core spaCy library. You can follow its development here: https://github.com/explosion/spaCy/pull/7264. You'll see that we've been actively working on this. Most of the code is written, but it needs some more debugging and fine-tuning.

I can't speak for HuggingFace and what they intend to do with `neuralcoref` going forward, ofcourse, and either way this will always be a useful library for the older versions of spaCy :-)

**Comment by KTRosenberg** [comment]
> Hi all, as a quick update that I also posted in a related thread.
> 
> While we (Explosion) originally intended to update `neuralcoref` in collaboration with HuggingFace (and @thomwolf had been very helpful with that), we've recently revamped spaCy to 3.0 and it turned out that quite a significant rewrite was necessary to support a native Thinc model in `neuralcoref`. Meanwhile also, research on coreference resolution had progressed further.
> 
> So long story short, we started working on a built-in coreference module that will be part of the core spaCy library. You can follow its development here: [explosion/spaCy#7264](https://github.com/explosion/spaCy/pull/7264). You'll see that we've been actively working on this. Most of the code is written, but it needs some more debugging and fine-tuning.
> 
> I can't speak for HuggingFace and what they intend to do with `neuralcoref` going forward and whether they're planning more updates, but either way this will always be a useful library for the older versions of spaCy :-)

That sounds good! Would a reasonable ETA be ~1 month for debugging and testing?

**Comment by svlandeg** [comment]
It's always difficult to provide these kind of estimates, as priorities sometimes shift, and we definitely want to "battle test" this new feature properly before we'll make it public.

**Comment by dogberto** [comment]
Awesome to hear this is happening! Thank you for putting the work into this. Please let us know when you think you have a release date :)

**Comment by KTRosenberg** [comment]
What is the status on this? Hopefully it's going well.

**Comment by nilesh** [comment]
Eagerly waiting for the core spacy3 integration. I can see that the pull request is out there and work in progress to merge. @KTRosenberg, you can track the progress here - https://github.com/explosion/spaCy/pull/7264

**Comment by KTRosenberg** [comment]
Oh I’m pretty sure they decided to work on a spacy coreference resolution feature in place of neuralcoref, if I understand correctly.

**Comment by danyang-rainbow** [comment]
Looking forward to see the spacy 3.0 or even 3.1 integration!✔

**Comment by grigvardanyan** [comment]
@svlandeg do you have any news :) ?

**Comment by svlandeg** [comment]
https://github.com/explosion/spaCy/pull/7264#issuecomment-986957673 ;-)

**Comment by qhungngo** [comment]
Looking forward to using it in SpaCy 3.0

**Comment by polm** [comment]
The new coreference component for spaCy 3 [has been released](https://explosion.ai/blog/coref).

**Comment by fakerybakery** [comment]
Hi @polm, personally, I think the neuralcoref library is much easier to use than the built-in coreference for SpaCy. Is it possible to make this library into a wrapper for the built-in coreference and preserve the same API?
Thank you!

**Comment by lordsoffallen** [comment]
Any Update on this one? Spacy has been keeping coref feature as experimental for 2 years now and there is not enough documentation to get it work. Would be nice to get this updated to v3 version
